tests/test_precommit_setup.py

```
@@ -0,0 +1,197 @@
+import sys
+from pathlib import Path
+
+import pytest
+import toml
+import yaml
+
+
+def _is_precommit_installed() -> bool:
+    """Helper to check if pre-commit is installed."""
+    try:
+        result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)